oio/cli/container/container.py

```diff
@@ -516,8 +525,28 @@ def take_action(self, parsed_args):
             listing = acct_meta['listing']
 
         columns = ('Name', 'Bytes', 'Objects', 'Mtime')
-        return columns, ((v['name'], v['bytes'], v['objects'], v['mtime'])
-                         for v in listing)
+
+        if parsed_args.versioning:
+            columns += ('Versioning',)
+            # TODO(mbo) rework to use benefits of yield if no-paging is enabled
+            data = None
+            for v in listing:
+                try:
+                    data = storage.container_get_properties(account, v['name'])
+                except NoSuchContainer:
+                    v['versioning'] = "Error"
+                    self.log.info('Bucket %s does not exist', v['name'])
+                    continue
+
+                sys = data['system']
+                # TODO(mbo) we should use default namespace configuration
```

Comment:
Enabling versioning by default on the whole namespace goes against S3 compatibility. This is a feature we should drop.